src/librustdoc/html/render/mod.rs

```diff
@@ -1598,7 +1583,7 @@ fn render_impl(
             outer_const_version,
         );
         write_srclink(cx, &i.impl_item, w);
-        if impl_items.is_empty() {
+        if impl_items.is_empty() && default_impl_items.is_empty() {
```

Comment:
Let's factor this statement out into a variable:

```
let toggled = impl_items.is_empty() && default_impl_items.is_empty();
if toggled {
```

Then we can use it below, too. That will make the relationship between the clauses more clear. Also, if we can set this variable high enough we can use it in the `open_details` closure too.